mcuboot: Use zephyr version when signing applications

nordicjm opened the pull request:
Removes the NCS-specific CONFIG_MCUBOOT_IMAGE_VERSION variable and uses the upstream zephyr CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION variable instead when signing images which use the MCUboot bootloader, this means that applications can now use an application VERSION file instead to specify their versions which will be automatically used when applications are signed instead of having to set the Kconfig value manually.

tejlmand reviewed (commented):
Looks good, great improvement.

How about documentation ?
Currently I don't see any docs or references to Zephyr docs in NCS that application's VERSION file is now used as input.
So far all usage of CONFIG_MCUBOOT_IMAGE_VERSION has been updtaed to CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION, but I don't see VERSION file description added.
Thus users can easily think that CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION is the only / correct way to do this.

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-12212/3/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-12212/3/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| desktop52_verification | X | | |
| test-ci-nrfconnect-boot-fw-update | X | | |
| test-fw-nrfconnect-apps | X | | |
| test-fw-nrfconnect-ble | X | | |
| test-fw-nrfconnect-ble_mesh | X | | |
| test-fw-nrfconnect-ble_samples | X | | |
| test-fw-nrfconnect-chip | X | | |
| test-fw-nrfconnect-tfm | X | | |
| test-fw-nrfconnect-zigbee | X | | |
| test-sdk-audio | X | | |
| test-sdk-find-my | X | | |
| test-sdk-sidewalk | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-12212/3/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

tejlmand commented:
> Needs a changelog entry but this is already documented upstream https://docs.zephyrproject.org/latest/build/version/index.html

I know it's described upstream, i'm just concerned that NCS developers who reads: 
- https://github.com/nrfconnect/sdk-nrf/blob/43d3500f5717be8dc05418661ce2c5ddff56b64a/doc/nrf/config_and_build/bootloaders_and_dfu/fw_update.rst
- https://github.com/nrfconnect/sdk-nrf/blob/43d3500f5717be8dc05418661ce2c5ddff56b64a/doc/nrf/config_and_build/bootloaders_and_dfu/fw_update.rst

which only mentions `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION `, will believe that `CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION ` is the only / official way in NCS to do this.

Shouldn't those docs mention or link to the app VERSION file doc ?

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-12212
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

frkv reviewed (approved):
LGTM (pending removal of private repos)

nordicjm commented:
Added a changelog entry which references the zephyr VERSION file documentation
Moved to using the open PR west.yml references
boot-fw-update tests need updating (have left a message)

Andrewpini reviewed (approved):
I've tested the mesh dfu samples with this commit, and it seems to run as expected 👍

nordicjm commented:
@hakonfam @MarekPieta @kapi-no @tomchy @Vge0rge @joerchan @SebastianBoe @mswarowsky @rlubos @juhaylinen @ludvigsj @koffes @rick1082 please take a look

wiba-nordic commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -53,7 +53,7 @@ RF Front-End Modules
 ------------
 
-|no_changes_yet_note|
+* Removed :kconfig:option:`CONFIG_MCUBOOT_IMAGE_VERSION` in favor of using a VERSION file, this version can be set via Kconfig by using :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION`, but should be transitioned to using a dedicated :ref:`application VERSION file <zephyr:app-version-details>` instead.
```
```suggestion
* Removed the :kconfig:option:`CONFIG_MCUBOOT_IMAGE_VERSION` Kconfig option in favor of using a dedicated :ref:`application VERSION file <zephyr:app-version-details>` file to set the version.
   You can alternatively set the version by using the :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` Kconfig option, but using a :file:`VERSION` file is the recommended approach.
```
I think that's still saying the same thing, just with a different focus.

nordicjm replied:
Updated

rick1082 reviewed (approved):
Test the DFU feature on Octave project, runs without issue.

wiba-nordic commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -53,7 +53,8 @@ RF Front-End Modules
 ------------
 
-|no_changes_yet_note|
+* Removed the ``CONFIG_MCUBOOT_IMAGE_VERSION`` Kconfig option in favor of using a dedicated :ref:`application VERSION file <zephyr:app-version-details>` file to set the version.
```
```suggestion
* Removed the ``CONFIG_MCUBOOT_IMAGE_VERSION`` Kconfig option in favor of using a dedicated :ref:`application VERSION file <zephyr:app-version-details>` to set the version.
```
Drat, I had a double "file" in there.

nordicjm replied:
Didn't notice that! Fixed

wiba-nordic reviewed (changes requested):
My own mistake that needs fixing.

joerchan commented on samples/tfm/tfm_psa_template/README.rst:
```diff
@@ -153,15 +153,15 @@ Application and TF-M firmware update
 ====================================
```
I can't verify that the changes of this PR to this sample is correct, because something is broken in main branch on signing keys.

joerchan reviewed (approved):
Tested the changes for the PSA template (with mcuboot patched as it is currently broken in main).

nordicjm commented on samples/bluetooth/peripheral_status/boards/thingy53_nrf5340_cpuapp_ns.conf:
```diff
@@ -55,7 +55,6 @@ CONFIG_HW_STACK_PROTECTION=y
 # Bootloader configuration
 
 CONFIG_BOOTLOADER_MCUBOOT=y
-CONFIG_MCUBOOT_IMAGE_VERSION="2.4.0+0"
```
It will actually retain the previous behaviour but needs https://github.com/zephyrproject-rtos/zephyr/pull/61852 brought in, which I will do after the upmerge

MarekPieta replied:
Just try to get it before the NCS 2.5. Otherwise the release note here may actually be needed.

nordicjm commented on doc/nrf/config_and_build/config_and_build_system.rst:
```diff
@@ -181,8 +181,8 @@ MCUboot output build files
 +------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
-| :file:`dfu_mcuboot.zip`                                          | Contains two versions of MCUboot linked against different address spaces corresponding to slot0 (s0) and slot1 (s1) and a manifest JSON file describing their MCUboot version number (``MCUBOOT_IMAGE_VERSION``), NSIB version number (``FW_INFO``), board type.   |
-|                                                                  | This file can be used by FOTA servers (for example, nRF Cloud) to serve both s0 and s1 to the device.                                                                                                                                                              |
+| :file:`dfu_mcuboot.zip`                                          | Contains two versions of MCUboot linked against different address spaces corresponding to slot0 (s0) and slot1 (s1) and a manifest JSON file describing their MCUboot version number (``MCUBOOT_IMGTOOL_SIGN_VERSION``), NSIB version number (``FW_INFO``), board  |
```
No clue on NSIB. @hakonfam ?
